<?xml version="1.0"?>
<launch>
	<param name="robot_description" command="$(find xacro)/xacro --inorder '$(find my_robot_description)/urdf/robot.urdf'" />

	<arg name="use_gui" default="true" doc="Should the joint_state_publisher use a GUI for controlling joint states" />
	<node pkg="joint_state_publisher" type="joint_state_publisher" name="joint_state_publisher" output="screen" unless="$(arg use_gui)" />
	<node pkg="joint_state_publisher_gui" type="joint_state_publisher_gui" name="joint_state_publisher_gui" output="screen" if="$(arg use_gui)" />

	<node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" respawn="false" output="screen" >
			<remap from="/joint_states" to="/my_robot/joint_states" />
	</node>

	<node name="rviz" pkg="rviz" type="rviz" args="-d $(find my_robot_description)/launch/config.rviz" />
</launch>

